pandas/sparse/frame.py

```diff
@@ -174,7 +171,36 @@ def _init_dict(self, data, index, columns, dtype=None):
         return to_manager(sdict, columns, index)
 
     def _init_matrix(self, data, index, columns, dtype=None):
+        """ Init self from ndarray or list of lists """
         data = _prep_ndarray(data, copy=False)
+        index, columns = self._prep_index(data, index, columns)
+        data = dict([(idx, data[:, i]) for i, idx in enumerate(columns)])
+        return self._init_dict(data, index, columns, dtype)
+
+    def _init_spmatrix(self, data, index, columns, dtype=None):
+        """ Init self from scipy.sparse matrix """
+        index, columns = self._prep_index(data, index, columns)
+        data = data.tocoo(copy=False)
```

Comment:
Works for any `scipy.sparse.spmatrix`. As per docs, it makes a copy if it needs to, but this is a copy of the sparse data, which is relatively cheap. Sparse matrix methods in SciPy do the conversion to the optimal type for the operation internally as needed, e.g. see `coo_matrix.maximum`, `lil_matrix.getcol`, `dok_matrix.__mul__` via `spmatrix._mul_sparse_matrix` ...

I am noting the transparent conversion (and potential copy) in the docs.